### PR TITLE
replace detached spin thread with std::jthread

### PR DESCRIPTION
**Pull request by ahcorde**
replaced teleop's detached spin thread and made running atomic, fixing a shutdown use-after-free race and signal-handle

@asymingt FYI

Claude Opus 4.7

### CODE REVIEW

**Comment by ahcorde** [comment]
friendly ping @asymingt

File: turtlesim/tutorials/teleop_turtle_key.cpp [hunk 1]
```diff
@@ -200,7 +202,17 @@ class TeleopTurtle final
   {
     char c;
 
-    std::thread{std::bind(&TeleopTurtle::spin, this)}.detach();
+    // Spin the node on a background thread. std::jthread joins automatically
```
Comment (asymingt) [inline turtlesim/tutorials/teleop_turtle_key.cpp:205]: Claude is explaining the change it made to *you* via this comment. To be committed into the code it needs to add texture to what's implemented in the current code revision. So, please remove the `unlike the previous detach() which raced shutdown` from this comment.

**Comment by ahcorde** [comment]
Pulls: ros/ros_tutorials#200
Gist: https://gist.githubusercontent.com/ahcorde/7e788c4e49d220c8ea64bd42398ae58c/raw/1906cb233b10348884e9ef0f9cae8582df5aea13/ros2.repos
BUILD args:  --packages-above-and-dependencies turtlesim
TEST args:  --packages-above turtlesim
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/19652
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/29509/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/22296/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/9398/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/28459/)

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/29516/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/22303/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/9405/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/28466/)

**Review by asymingt — APPROVED** [review]
LGTM

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/29524/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/22311/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/9413/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/28476/)
